src/test/scala/latis/input/HapiAdapterSpec.scala

```diff
@@ -0,0 +1,57 @@
+package latis.input
+
+import java.net.URI
+import org.scalatest.FlatSpec
+import org.scalatest.Matchers._
+
+import latis.data._
+import latis.dataset._
+import latis.metadata.Metadata
+import latis.model._
+import latis.ops.Selection
+import latis.util.StreamUtils
+
+object HapiAdapterSpec extends FlatSpec {
+  
+  val model = Function(
+    Scalar(Metadata(
+      "id"   -> "time",
+      "type" -> "string"
+    )),
+    Scalar(Metadata(
+      "id"   -> "tsi_1au",
+      "type" -> "double"
+    ))
+  )
+  
+  val adapter = new HapiCsvAdapter(
+    model,
+    new HapiAdapter.Config(
+      "class" -> "latis.input.HapiCsvAdapter",
+      "id"    -> "sorce_tsi_24hr_l3"
+    )
+  )
+  
+  val baseUri = new URI("http://lasp.colorado.edu/lisird/hapi/")
+  
+  val dataset = new AdaptedDataset(
+    Metadata("sorce_tsi"),
+    model,
+    adapter,
+    baseUri
+  )
+//  val dataset = Dataset.fromName("sorce_tsi") //get from fdml
```

Comment:
Any thoughts on defining the dataset in code vs using fdml?

Reply:
Sorry, I haven't actually used an FDML before. What would it look like defined in FDML?

Reply:
Very similar to tsml: https://github.com/latis-data/latis3/blob/master/core/src/test/resources/datasets/data.fdml
I'm just not sure if we want a bunch of test fdml files floating around or to define test datasets in code closer to the test.

Reply:
If it's just for the tests then I lean toward defining the dataset in code. That way when I'm going to look at a test, I don't also have to look in an FDML to see the dataset.

Reply:
If it's just for the tests then I lean toward defining the dataset in code. That way when I'm going to look at a test, I don't also have to look in an FDML to see the dataset.